webcompat/api/uploads.py

```diff
@@ -100,6 +106,10 @@ def save(self):
             save_parameters['save_all'] = True
         # unpacking save_parameters
         self.image_object.save(file_dest, **save_parameters)
+        # Creating the thumbnail
+        size = (150, 150)
+        self.image_object.thumbnail(size, Image.ANTIALIAS)
```

Comment:
@miketaylr got time to test a bit more.

First the results.

```
→ python resize-filter.py 

    thumb_nearest.jpg:
        thumbnail only   0.01 seconds
        thumbnail + save 0.03 seconds
    

    thumb_bilinear.jpg:
        thumbnail only   0.03 seconds
        thumbnail + save 0.04 seconds
    

    thumb_bicubic.jpg:
        thumbnail only   0.04 seconds
        thumbnail + save 0.05 seconds
    

    thumb_lanczos.jpg:
        thumbnail only   0.05 seconds
        thumbnail + save 0.06 seconds
    

    thumb_antialias.jpg:
        thumbnail only   0.05 seconds
        thumbnail + save 0.06 seconds
```
    
```
→ ls -al
total 1976
drwxr-xr-x  9 karl  staff     306 12 oct 15:09 .
drwxr-xr-x  4 karl  staff     136 12 oct 15:05 ..
-rw-r--r--@ 1 karl  staff  826406 12 oct 14:16 original.png
-rw-r--r--  1 karl  staff    1023 12 oct 15:09 resize-filter.py
-rw-r--r--  1 karl  staff   33618 12 oct 15:09 thumb_antialias.jpg
-rw-r--r--  1 karl  staff   32524 12 oct 15:09 thumb_bicubic.jpg
-rw-r--r--  1 karl  staff   30179 12 oct 15:09 thumb_bilinear.jpg
-rw-r--r--  1 karl  staff   33618 12 oct 15:09 thumb_lanczos.jpg
-rw-r--r--  1 karl  staff   38890 12 oct 15:09 thumb_nearest.jpg
```

The script to test it.

```python
from PIL import Image
import time

FILTER_LIST = [('nearest', Image.NEAREST),
               ('bilinear', Image.BILINEAR),
               ('bicubic', Image.BICUBIC),
               ('lanczos', Image.LANCZOS),
               ('antialias', Image.ANTIALIAS)]
FILE = '/Users/karl/Documents/2016/10/12/thumbnail-test/original.png'
SIZE = (500, 500)

original = Image.open(FILE)

for thumb_name, img_filter in FILTER_LIST:
    # We put the cursor at the start
    original2 = original.copy()
    thumb_name = 'thumb_{}.jpg'.format(thumb_name)
    # Creating the thumbnail
    start = time.time()
    original2.thumbnail(SIZE, img_filter)
    middle = time.time()
    original2.save(thumb_name, 'JPEG')
    end = time.time()
    duration = round(end - start, 2)
    mid_duration = round(middle - start, 2)
    print('''
    {thumb_name}:
        thumbnail only   {mid_duration} seconds
        thumbnail + save {duration} seconds
    '''.format(thumb_name=thumb_name, duration=duration,
               mid_duration=mid_duration))
```

Resized to be 500px. 
Original image is a PNG screenshot of JapanTimes Web site `1366x768 px` ~ 807K

but because I spent a bit of time in the internals of Pillow a couple of months ago. I know they use a lot the local libraries for images. So I would recommend you do the tests on the server itself. Just to see if we have consistent results.